core/src/main/scala/akka/kafka/internal/DefaultProducerStage.scala

```diff
@@ -99,21 +103,28 @@ private class DefaultProducerStageLogic[K, V, P, IN <: Envelope[K, V, P], OUT <:
     }
   }
 
-  protected def suspendDemand(): Unit =
+  protected def suspendDemand(fromStageLogicConstructor: Boolean = false): Unit = {
+    // not permitted to access stage logic members from constructor
+    if (!fromStageLogicConstructor) log.debug("Suspend demand")
```

Comment:
I agree it's weird.  Maybe we can suspend demand directly in the constructor instead of calling this method.  I think the logging under normal circumstances is useful.

Reply:
I factored out setting the `OutHandler` so it can be called from `suspendDemand` and the stage logic constructor.